apps/builder/app/builder/features/breakpoints/use-update-canvas-width.ts

```diff
@@ -3,21 +3,27 @@ import { useCallback, useEffect } from "react";
 import { useCanvasWidth } from "~/builder/shared/nano-states";
 import { useIsPreviewMode } from "~/shared/nano-states";
 import { selectedBreakpointStore } from "~/shared/nano-states/breakpoints";
+import { useNextBreakpoint } from "./use-next-breakpoint";
 import { minWidth } from "./width-setting";
 
 export const useUpdateCanvasWidth = () => {
   const selectedBreakpoint = useStore(selectedBreakpointStore);
   const [canvasWidth, setCanvasWidth] = useCanvasWidth();
   const [isPreviewMode] = useIsPreviewMode();
+  const nextBreakpoint = useNextBreakpoint();
 
   // Ensure the size is within currently selected breakpoint when returning to design mode out of preview mode,
   // because preview mode enables resizing without constraining to the selected breakpoint.
   useEffect(() => {
     if (isPreviewMode || selectedBreakpoint === undefined) {
       return;
     }
-    setCanvasWidth(Math.max(selectedBreakpoint.minWidth, minWidth));
-  }, [isPreviewMode, selectedBreakpoint, setCanvasWidth]);
+    const width = Math.max(
```

Comment:
At this point, we can measure the width of iframe parent and use it here to not set 767px when awailable width is less

Reply:
done